src/Runner.Worker/FileCommandManager.cs

```diff
@@ -182,35 +185,75 @@ public void ProcessCommand(IExecutionContext context, string filePath, Container
                     return;
                 }
 
-                if (fileSize > AttachmentSizeLimit)
-                {
-                    context.Error(String.Format(Constants.Runner.UnsupportedSummarySize, AttachmentSizeLimit / 1024, fileSize / 1024));
-                    Trace.Info($"Step Summary file ({filePath}) is too large ({fileSize} bytes); skipping attachment upload");
+                Trace.Verbose($"Step Summary file exists: {filePath} and has a file size of {fileSize} bytes");
+                if (context.Global.Variables.Get("ResultsReceiverEndpoint") != null) {
+                    Trace.Info("Using new Actions Results Job Summaries upload");
+
+                    var ResultsReceiverEndpointAddr = context.Global.Variables.Get("ResultsReceiverEndpoint");
+
+                    var token = context.Global.Variables.Get("System.AccessToken");
+                    var resultsClient = new ResultsHttpClient(token, ResultsReceiverEndpointAddr);
+                    var planId = context.Global.Variables.Get("System.PlanId");
+                    var jobId = context.Global.Variables.Get("System.JobId");
+                    var stepId = context.Global.Variables.Get("System.StepId");
+                    var stepSummaryUploadUrl = await resultsClient.GetStepSummaryUploadUrlAsync(jobId, planId, stepId, new CancellationToken());
+                    
+                    // Do file size check
+                    if (fileSize > stepSummaryUploadUrl.Size)
+                    {
+                        context.Error(String.Format(Constants.Runner.UnsupportedSummarySize, stepSummaryUploadUrl.Size / 1024, fileSize / 1024));
+                        Trace.Info($"Step Summary file ({filePath}) is too large ({fileSize} bytes); skipping attachment upload");
 
-                    return;
-                }
+                        return;
+                    }
 
-                Trace.Verbose($"Step Summary file exists: {filePath} and has a file size of {fileSize} bytes");
-                var scrubbedFilePath = filePath + "-scrubbed";
+                    // Upload the file to the url
+                    var httpClient = new HttpClient();
+                    var request = new HttpRequestMessage(HttpMethod.Put, stepSummaryUploadUrl.SummaryUrl);
+                    // Get response with the storage type, set the correct headers for Azure vs. S3 (only worry about Azure for now)
+                    using (var response = await httpClient.SendAsync(request, HttpCompletionOption.ResponseHeadersRead)) {
```

Comment:
I'm wondering if we could do this in a background process similar to how we're submitting job summaries today 🤔 

```c#
context.Root.QueueAttachFile(ChecksAttachmentType.StepSummary, attachmentName, scrubbedFilePath);
```

I'm not sure if all the context needed to successfully upload summaries will be available from the background thread that's being used here but it might be worth a shot.

Reply:
Please do the background way. 🙇